src/library/tools/R/check.R

```diff
@@ -3950,6 +3950,22 @@ add_dummies <- function(dir, Log)
             msg <- c(msg, "No rustc version reported prior to compilation")
 ##            print(lines)
         }
+        desc <- .read_description("DESCRIPTION")
+        sysreqs <- parse_description_field(desc, "SystemRequirements", default = FALSE)
+        if(!sysreqs) {
+            OK <- FALSE
+            msg <- c(msg, "Package compiles Rust but SystemRequirements not provided")
+        }
+        rustc_desc <- any(grep("rustc", sysreqs, ignore.case = TRUE))
+        cargo_desc <- any(grep("cargo", sysreqs, ignore.case = TRUE))
```

Comment:
A small suggestion, I don't really know if it matters but might help to clarify the intention:

```suggestion
        rustc_desc <- any(grep("rustc", sysreqs, ignore.case = TRUE, fixed = TRUE))
        cargo_desc <- any(grep("cargo", sysreqs, ignore.case = TRUE, fixed = TRUE))
```